test-framework/common/src/main/java/io/quarkus/test/common/TestResourceScope.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.test.common;
+
+/**
+ * Defines how Quarkus behaves with regard to the application of the resource to this test and the testsuite in general
+ */
+public enum TestResourceScope {
+
+    /*
+     * The declaration order must be from the narrowest scope to the widest
+     */
+
+    /**
+     * Means that Quarkus will run the test in complete isolation, i.e. it will restart every time it finds such a resource
+     */
+    RESTRICTED_TO_CLASS,
+    /**
+     * Means that Quarkus will not restart when running consecutive tests that use the same resource
```

Comment:
The same resource or the same set of resources? If the former how do you group to restart the resources less often?

Reply:
Users can't really influence the order of test execution, but if Quarkus does its job correctly, restarting will be minimized

Reply:
How? (for my benefit).

Reply:
https://github.com/quarkusio/quarkus/blob/dcd9928aed836f1b2b4d4a2274bef1b767a97311/test-framework%2Fjunit5%2Fsrc%2Fmain%2Fjava%2Fio%2Fquarkus%2Ftest%2Fjunit%2Futil%2FQuarkusTestProfileAwareClassOrderer.java

We might need to update it a little, but the idea is that we can tell JUnit the order with which tests can be executed

Reply:
Also my question was
`@WithTestResource(value={Resource1.class, Resource2.class, Resource3.class})`
will in run in the same batch as `@WithTestResource(value={Resource3.class})`
what about `@WithTestResource(value={Resource3.class, Resource4.class})`

Reply:
Even if those are executed one after the other, they still force a restart because they don't match

Reply:
ok, thanks. So i'ts not the same resource that matches, it's the exact resource set (or list of resoruces) that matches.

Reply:
Correct!